backend/fms_core/template_importer/row_handlers/project_link_samples/project_link_samples.py

```diff
@@ -28,18 +35,75 @@ def process_row_inner(self, sample, project, action):
                               f"is named {sample_obj.name} not {sample['sample_name']}."
                 self.warnings['sample'].append(warning_msg)
 
-            # Check if link exists to ensure there's not a duplicated association
-            if action['name'] == ADD_ACTION:
+            # Perform an add project and/or study action
+            if action['name'] == ADD_PROJECT_STUDY_ACTION:
                 # Create link object if no errors
                 link_created, self.errors['link'], self.warnings['link'] = create_link(sample=sample_obj,
                                                                                        project=project_obj)
 
-            # If the link doesn't exists we can't perform a remove action
-            elif action['name'] == REMOVE_ACTION:
+                # Queue sample to study if specified
+                if project['study_letter']:
+                    study_obj, self.errors['study'], self.warnings['study'] = get_study(project_obj, project['study_letter'])
+
+                    if study_obj:
+                        study_start = project['study_start']
+
+                        # Check if sample is already queued 
+                        is_sample_queued, self.errors['sample_queued'], self.warnings['sample_queued'] = is_sample_queued_in_study(sample_obj, study_obj, study_start)
+                        if not is_sample_queued:
+                            _, self.errors['add_to_study'], self.warnings['queue_to_study'] = queue_sample_to_study_workflow(sample_obj, study_obj, study_start)
+                        else:
+                            self.errors['study'] = f"Sample [{sample_obj.name}] is already queued in study [{project['study_letter']}] \
```

Comment:
you can have a sample queued to multiple steps of the study. This is not an error unless you are adding the same sample at the same step order of the study, If the study_start is empty you'll get an error which should not be the case unless the sample is queued on the first step of the study workflow.